### ISSUE TITLE
Question about usage of VFI model

### ISSUE THREAD

**Issue by Qianqian3764** [LiuJF1226/Mono-ViFI#3]
Hi, thanks for your great job！

For TA method，why use VFI model to generate intermediate frames? Isn't using intermediate frames as labels when training VFI models? My question is, since the middle frame can be obtained, why do we need to use an additional VFI model to generate it?
If the VFI model serves the subsequent multi frame feature fusion, it is reasonable because it needs to be used to generate the optical flow from the target frame to the source frame.

**Comment by LiuJF1226** [comment]
Thank you for your attention! The data split used in VFI training is identical to that employed in self-supervised MDE training. In standard self-supervised MDE training, a data sample consists of three consecutive frames ($I_{t-1}$, $I_{t}$, $I_{t+1}$), where $I_{t}$ is the target frame and the other two are source frames. **In VFI training, we interpolate $I_{t}$ using $I_{t-1}$ and $I_{t+1}$, with $I_{t}$ as the label, and no other intermediate frames are utilized as labels.** Once the VFI model is trained,, we utilize ($I_{t-1}$, $I_{t}$) to synthesize an intermediate frame $I_{t-0.5}$, and ($I_{t}$, $I_{t+1}$) to synthesize  $I_{t+0.5}$. $I_{t-0.5}$ and $I_{t+0.5}$ serve as two additional target frames in our following self-supervisef MDE training, for both single-frame and multi-frame depth models. This diversifies data distribution in terporal dimension. Also, the VFI model serves the subsequent multi frame feature fusion as you said.

**Comment by Qianqian3764** [comment]
Thanks very much for your reply！ 
This question is completely dispelled. That's a great idea!
I looked at the training code train.py. 

![image](https://github.com/user-attachments/assets/7361fe9e-2022-4344-9d3f-461ea66f6ef2)


Why not count the loss of [-0.5,0,0.5] for a single frame?

**Comment by LiuJF1226** [comment]
The core is utilizing -0.5 and 0.5 as target frames. Since a target frame is the input to depth model, novel target frames can enhance the model's capacity (depth model can see more samples). However, using -0.5 and 0.5 as sources will do not make too much sense, as the camera motion usually contains only translation component in autonomous driving scenes (KITTI and Cityscapes). In other words, the effect of -0.5/0.5 will overlap largely with -1/1 as source frames. That is why we usually consider a triple -1/0/1 instead of a quintuple -2/-1/0/1/2 in KITTI/Cityscapes training. In contrast, for indoor scenes like NYUv2, a common setting is -2/-1/0/1/2 (4 source frames). Since the camera motion is more complicated in indoor scenes, including both rotation and translation, more source frames will provide more information. While our work mainly focuses on autonomous driving scenes, it's also interesting to explore the application on indoor scenes.

Additionally, we have also tried that setting you said. It brings no performance gains but increases training cost.

**Comment by Qianqian3764** [comment]
Thank you very much for your detailed reply! This question is completely addressed. 

For multi-frame (MF), the way to integrate the temporal information is to convert the feature map of the source frame to the target frame through optical flow, and finally fuse. Because in the end are both the fused feature of the target frame, why will it have an effect? Individual target frame features do not work. 

Another problem is how to distinguish the specific use of mask, for example, M1 goes with I(0->1), (1-M1) goes with I(2->1).
![image](https://github.com/user-attachments/assets/a5daa2e6-b195-4d4f-b494-b1780fe432ec)

**Comment by LiuJF1226** [comment]
Consider the same semantic point in the 3 neighbor frames, its values in the feature maps of the 3 frames are different due to different image contexts. Therefore, multi-frame fusion feature actually provides more information than individual target frame feature. Feature fusion can aggregate temporal information to implicitly infer scene depth, where:
1) feature alignment via optical flow provides spatial relevance among the same location in neighbor frames.
2) the aligned features are concatenated with encoded optical flow, which indicates the relative coordinate of a feature vector in the warped feature map with respect to the original one. Injecting pixel motion (optical flow) to feature embeddings implicitly provides relative distance relation. Larger pixel motion usually indicates closer depth. Although this is not suitable for dynamic pixels, the influence is slight since this information is implicit.
3) the merge mask can alleviate misleading occlusion pixels.

The merge mask $M_1$ is also a intermediate product of our VFI model ([IFRNet](https://arxiv.org/pdf/2205.14620)), which reflects the bidirectional occlusion information. $M_1$ is pixel-aligned with target $I_1$. In this equation, if $M_1=0$ and $1-M_1=1$ for a pixel in $I_1$, then its corresponding pixel in $I_0$ is occluded and that in $I_2$ is visible, and vice versa.

**Comment by Qianqian3764** [comment]
Thank you again for your detailed answer, which has helped me further understand the multi-frame method of fully utilizing neighboring frames. I think that sometimes the ref features may not be accurate, and warped ref features can make up for it well.
